test/opt/redundancy_elimination_test.cpp

```diff
@@ -272,6 +272,59 @@ TEST_F(RedundancyEliminationTest, KeepRedundantAddWithoutPhi) {
   EXPECT_EQ(Pass::Status::SuccessWithoutChange, std::get<1>(result));
 }
 
+// Test that it can get a simple case of local redundancy elimination
+// when it has OpenCL.DebugInfo.100 DebugDeclare/DebugValue instructions.
+TEST_F(RedundancyEliminationTest, DebugValue) {
+  const std::string text = R"(
+               OpCapability Shader
+          %1 = OpExtInstImport "OpenCL.DebugInfo.100"
+          %2 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
+               OpEntryPoint Fragment %3 "main"
+               OpExecutionMode %3 OriginUpperLeft
+               OpSource GLSL 430
+          %4 = OpString "ps.hlsl"
+          %5 = OpString "float"
+          %6 = OpString "s0"
+          %7 = OpString "main"
+       %void = OpTypeVoid
+          %9 = OpTypeFunction %void
+      %float = OpTypeFloat 32
+       %uint = OpTypeInt 32 0
+     %uint_0 = OpConstant %uint 0
+    %uint_32 = OpConstant %uint 32
+%_ptr_Function_float = OpTypePointer Function %float
+         %15 = OpExtInst %void %1 DebugExpression
+         %16 = OpExtInst %void %1 DebugSource %4
+         %17 = OpExtInst %void %1 DebugCompilationUnit 1 4 %16 HLSL
+         %18 = OpExtInst %void %1 DebugTypeBasic %5 %uint_32 Float
+         %19 = OpExtInst %void %1 DebugTypeVector %18 4
+         %20 = OpExtInst %void %1 DebugTypeFunction FlagIsProtected|FlagIsPrivate %19
+         %21 = OpExtInst %void %1 DebugFunction %7 %20 %16 4 1 %17 %7 FlagIsProtected|FlagIsPrivate 4 %3
+         %22 = OpExtInst %void %1 DebugLocalVariable %6 %19 %16 0 0 %21 FlagIsLocal
+          %3 = OpFunction %void None %9
+         %23 = OpLabel
+         %24 = OpExtInst %void %1 DebugScope %21
+         %25 = OpVariable %_ptr_Function_float Function
+         %26 = OpLoad %float %25
+               OpLine %4 0 0
+; CHECK:      [[add:%\w+]] = OpFAdd %float [[value:%\w+]]
+; CHECK-NEXT: DebugValue {{%\w+}} [[add]]
+; CHECK-NEXT: OpFAdd %float [[add]] [[value]]
```

Comment:
It is not clear that the section OpFAdd corresponds to the one just before the return.  A comment might be useful.

Reply:
done